scripts/rabbitmq-env

```diff
@@ -19,334 +19,510 @@ if [ "$RABBITMQ_ENV_LOADED" = 1 ]; then
+    [ -n "$RABBITMQ_ALLOW_INPUT" ] || _rmq_env_noinput_arg='-noinput'
+    [ -n "$RABBITMQ_NODE_ONLY" ] || _rmq_env_node_only_arg="-s $RABBITMQ_BOOT_MODULE boot"
+    [ -n "$_rmq_env_noinput_arg" ] && [ -n "${_rmq_env_node_only_arg}" ] && _rmq_env_maybe_space=' '
+
+    # shellcheck disable=SC2034
+    _rmq_env_start_rabbit="${_rmq_env_noinput_arg}${_rmq_env_maybe_space}${_rmq_env_node_only_arg}"
+
+    unset _rmq_env_noinput_arg
+    unset _rmq_env_node_only_arg
+    unset _rmq_env_maybe_space
+}
+
+_rmq_env_config_alloc_args()
+{
+    # The default allocation strategy RabbitMQ is using was introduced
+    # in Erlang/OTP 20.2.3. Earlier Erlang versions fail to start with
+    # this configuration. We therefore need to ensure that erl accepts
+    # these values before we can use them.
+    #
+    # The defaults are meant to reduce RabbitMQ's memory usage and help
+    # it reclaim memory at the cost of a slight decrease in performance
+    # (due to an increase in memory operations). These defaults can be
+    # overriden using the RABBITMQ_SERVER_ERL_ARGS variable.
+    # shellcheck disable=SC2034
+    #
+    # Note: the default value is
+    #   +MBas ageffcbf
+    #   +MHas ageffcbf
+    #   +MBlmbcs 512
+    #   +MHlmbcs 512
+    #   +MMmcs 30
+    [ -n "$ALLOC_ARGS" ] || ALLOC_ARGS="+MBas ageffcbf +MHas ageffcbf +MBlmbcs 512 +MHlmbcs 512 +MMmcs 30"
+    [ -n "$RABBITMQ_ALLOC_ARGS" ] || RABBITMQ_ALLOC_ARGS="$ALLOC_ARGS"
+}
+
+_rmq_env_config_mnesia_dir()
+{
+    [ -n "$RABBITMQ_MNESIA_DIR" ] || RABBITMQ_MNESIA_DIR="$MNESIA_DIR"
+    [ -n "$RABBITMQ_MNESIA_DIR" ] || RABBITMQ_MNESIA_DIR="$RABBITMQ_MNESIA_BASE}/$RABBITMQ_NODENAME"
+    _rmq_env_mnesia_dir_configured='true'
+}
+
+_rmq_env_config_pid_file()
+{
+    [ "$_rmq_env_mnesia_dir_configured" = 'true' ] || _rmq_env_errexit '_rmq_env_config_mnesia_dir must be called prior to _rmq_env_config_pid_file'
+    unset _rmq_env_mnesia_dir_configured
+
+    [ -n "$RABBITMQ_PID_FILE" ] || RABBITMQ_PID_FILE="$PID_FILE"
+    [ -n "$RABBITMQ_PID_FILE" ] || RABBITMQ_PID_FILE="$RABBITMQ_MNESIA_DIR.pid"
+    _rmq_env_normalize_path_var RABBITMQ_PID_FILE
+
+    # We save the current value of $RABBITMQ_PID_FILE in case it was set by
+    # an init script. If $CONF_ENV_FILE overrides it again, we must ignore
+    # it and warn the user.
+    _rmq_env_saved_RABBITMQ_PID_FILE="$RABBITMQ_PID_FILE"
+
+    if [ -n "$_rmq_env_saved_RABBITMQ_PID_FILE" ] && \
+        [ "$_rmq_env_saved_RABBITMQ_PID_FILE" != "$RABBITMQ_PID_FILE" ]; then
+        echo "WARNING: RABBITMQ_PID_FILE was already set by the init script to:" 1>&2
+        echo "           $_rmq_env_saved_RABBITMQ_PID_FILE" 1>&2
+        echo "         The value set in rabbitmq-env.conf is ignored because it" 1>&2
+        echo "         would break the init script." 1>&2
+
+        RABBITMQ_PID_FILE="$_rmq_env_saved_RABBITMQ_PID_FILE"
+    fi
+    unset _rmq_env_saved_RABBITMQ_PID_FILE
+}
+
+_rmq_env_TODO()
```

Comment:
Do you want to merge with this function?
Should it be renamed then?
It looks that some variables are set in other functions.

Reply:
We can merge this now, and then we'll be working on the same branch (which is A-OK), or I can get all the TODOs done and then merge. Just let me know.